cfme/tests/infrastructure/test_webmks_console.py

```diff
@@ -0,0 +1,169 @@
+def configure_vmware_console_for_test(appliance, provider):
+    """Configure VMware Console to use VNC which is what is required for the HTML5 console."""
+    navigate_to(appliance.server, 'Server')
+
+    settings_pg = VMwareConsoleSupport(
+        appliance=appliance,
+        console_type='VMware WebMKS',
+    )
+    settings_pg.update()
+
+
+@pytest.yield_fixture(scope="session")
+def configure_websocket():
+    """Enable websocket role if it is disabled.
+
+    Currently the fixture cfme/fixtures/base.py,
+    disables the websocket role to avoid intrusive popups. It is enabled
+    here before test begins and disabled during teardown.
+    """
+    roles = configuration.get_server_roles()
+    if 'websocket' in roles and not roles['websocket']:
+        logger.info('Enabling the websocket role to allow console connections')
+        roles['websocket'] = True
+        configuration.set_server_roles(**roles)
+    yield
+    roles['websocket'] = False
+    logger.info('Disabling the websocket role to avoid intrusive popups')
+    configuration.set_server_roles(**roles)
+
+
+@pytest.mark.uncollectif(lambda: version.current_version() < '5.8', reason='Only valid for >= 5.8')
+def test_html5_vm_console(request, appliance, provider, vm_obj, configure_websocket,
+        configure_vmware_console_for_test):
+    """Test the HTML5 console support for a particular provider.
+
+    The supported providers are:
+
+        VMware
+        Openstack
+        RHV
+
+    For a given provider, and a given VM, the console will be opened, and then:
+
+        - The console's status will be checked.
+        - A command that creates a file will be sent through the console.
+        - Using ssh we will check that the command worked (i.e. that the file
+          was created.
+    """
+    console_vm_username = credentials[provider.data.templates.get('console_template')
+                            ['creds']].get('username')
+    console_vm_password = credentials[provider.data.templates.get('console_template')
+                            ['creds']].get('password')
+
+    vm_obj.open_console(console='VM Console', invokes_alert=True)
+    assert vm_obj.vm_console, 'VMConsole object should be created'
+    vm_console = vm_obj.vm_console
+
+    request.addfinalizer(vm_console.close_console_window)
+    request.addfinalizer(appliance.server.logout)
+
+    # Get the login screen image, and make sure it is a jpeg file:
+    screen = vm_console.get_screen(180)
+    assert imghdr.what('', screen) == 'jpeg'
+
+    assert vm_console.wait_for_text(text_to_find="login:", timeout=200),\
+        "VM Console didn't prompt for Login"
+
+    # Enter Username:
+    vm_console.send_keys(console_vm_username)
+
+    assert vm_console.wait_for_text(text_to_find="Password", timeout=200),\
+        "VM Console didn't prompt for Password"
+    # Enter Password:
+    vm_console.send_keys("{}\n".format(console_vm_password))
+
+    time.sleep(5)  # wait for login to complete
```

Comment:
@quarckster  There is not really a way to know this, time.sleep is needed because after I enter username and password, I need to wait at least a sec or two before I get either command prompt or "Login Incorrect" message. So I can not think of anything besides time.sleep in this situation. You might say I can wait_for _command prompt_ or _"Login Incorrect"_ message. But for Image to Text recognition finding out command prompt correctly is a challenge. And if I detect _"Login Incorrect"_ I will then need to attempt to login again. Hence I just replaced all that login with validate_login which validates if login went through and if not, attempt to re-login. Any suggestions to make it even better are welcome. Thanks :)